packages/core/src/App/Components/Layout/Header/account-actions.jsx

```diff
@@ -38,19 +43,27 @@ const AccountActions = React.memo(
         toggleAccountsDialog,
         toggleNotifications,
         is_deposit_button_disabled,
+        is_mobile,
     }) => {
         const { is_appstore } = React.useContext(PlatformContext);
 
-        if (is_logged_in) {
-            return (
+        const { is_wallet_enabled } = useFeatureFlags();
+        const { has_wallet, isLoading } = useWalletAccountsList();
+
+        const should_show_wallets = is_wallet_enabled && has_wallet;
```

Comment:
@sergei-deriv @arshad let's add two seperate files for `wallets` and `without_wallets` account switcher instead of adding condition here and then in this file decide which one needs to be shown here based on the availability of wallets 
```suggestion
   return(
   should_show_wallets ? <walletsSwitcher/> : <withoutWalletsSwitcher/>
   )
```

Reply:
@mahdiyeh-deriv I'm little mixed up with your suggestion where I need to create this switcher. We have the next structure: Header -> DTraderHeader -> AccountActions -> AccountInfo (shows selected account in the header) -> AccountSwitcher (dropdown for accounts). We've created `AccountInfoWallets` component and `AccountSwitcherWallets` and check to show wallets or not in `AccountActions` component. Where do you want we check for wallet, create `CommonDTraderHeader` and switch components there, or maybe create `CommonAccountActions` and switch components there, or create `CommonAccountInfo` and switch components there.